homeassistant/components/mystrom/switch.py

```diff
@@ -67,6 +67,11 @@ def current_power_w(self):
         """Return the current power consumption in W."""
         return self.plug.consumption
 
+    @property
+    def current_temperature_c(self):
```

Comment:
This isn't used anywhere. Temperature should be a separate sensor entity.

Reply:
It's exactly like `current_power_w`. 

And with a sensor template:
```    
sensor:
- platform: template
  sensors:
    office_temp:
      friendly_name: Office Temp
      unit_of_measurement: 'ºC'
      value_template: "{{ state_attr('switch.office_lamp', 'current_temperature_c') }}"
```

@MartinHjelmare Do you have an example of how it should be done better?

Reply:
The temperature measurement should be exposed as a separate sensor entity in a sensor platform. This first requires a separate refactor of this integration to use config entries and config flow.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler